crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -231,14 +239,23 @@ impl<T: TransactionOrdering> TxPool<T> {
 
                 Ok(res)
             }
-            Err(InsertErr::Underpriced { existing, .. }) => {
-                Err(PoolError::ReplacementUnderpriced(existing))
-            }
-            Err(InsertErr::ProtocolFeeCapTooLow { transaction, fee_cap }) => {
-                Err(PoolError::ProtocolFeeCapTooLow(*transaction.hash(), fee_cap))
-            }
-            Err(InsertErr::ExceededSenderTransactionsCapacity { transaction }) => {
-                Err(PoolError::SpammerExceededCapacity(transaction.sender(), *transaction.hash()))
+            Err(e) => {
+                // Update invalid transactions metric
+                self.metrics.invalid_transactions.increment(1);
```

Comment:
Do we want to be more granular here and actually have a metric on a per error basis? Maybe not, just asking the Q.

Reply:
sure! I guess that tracking the exact error could be useful. I can make a separate PR for that